test/Esprima.Tests.SourceGenerators/Snapshots/StringMatcherGeneratorTests.ScannerStringMatchingGeneration.01StringMatchers.g.verified.cs

```diff
@@ -215,91 +215,83 @@ public partial class Scanner
     private static partial string? TryGetInternedTwoCharacterPunctuator(ReadOnlySpan<char> input)
     {
         var c1 = input[1];
-        if (input[0] == '&')
+        if (input[0] == '-')
```

Comment:
nit: Any chance that we can have the nice, readable switch syntax here instead of if-else-ifs as it's done in some other cases? E.g.:

```csharp
return input[0] switch
{
    '-' => input[1] switch
    {
        '-' => return "--",
        '=' => return "-=",
         _ => null
    },
    // ...
}
```